src/solver/conv_hip_implicit_gemm_wrw_weights_v4r4.cpp

```diff
@@ -0,0 +1,782 @@
+        config = {128, 64, 128, 8, 4, 4};
+    if(!config.IsValid(ctx))
+        config = {128, 64, 128, 4, 4, 4};
+    if(!config.IsValid(ctx))
+        config = {64, 64, 64, 16, 4, 4};
+    if(!config.IsValid(ctx))
+        config = {64, 64, 64, 8, 4, 4};
+    if(!config.IsValid(ctx))
+        config = {64, 64, 64, 4, 4, 4};
+    if(!config.IsValid(ctx))
+        config = {64, 64, 32, 16, 4, 2};
+    if(!config.IsValid(ctx))
+        config = {64, 64, 32, 8, 4, 2};
+    if(!config.IsValid(ctx))
+        config = {64, 64, 32, 4, 4, 2};
+    if(!config.IsValid(ctx))
+        config = {64, 32, 64, 16, 2, 4};
+    if(!config.IsValid(ctx))
+        config = {64, 32, 64, 8, 2, 4};
+    if(!config.IsValid(ctx))
+        config = {64, 32, 64, 4, 2, 4};
+    if(!config.IsValid(ctx))
+        config = {64, 32, 32, 16, 2, 2};
+    if(!config.IsValid(ctx))
+        config = {64, 32, 32, 8, 2, 2};
+    if(!config.IsValid(ctx))
+        config = {64, 32, 32, 4, 2, 2};
+    if(!config.IsValid(ctx))
+    {
+        MIOPEN_LOG_E("All attempts failed: ");
+        assert(false);
+    }
+
+    *this = config;
+    MIOPEN_LOG_I(ToString());
+}
+
+bool PerformanceImplicitGemmV4R4WrW::SetNextValue()
+{
+    // always search full space, no matter if use_spare_set or not
+    do
+    {
+        if(!NextTwoPower<64, 256>(BlockSize))
+            break;
+        if(!NextTwoPower<32, 128>(GemmMPerBlock))
+            break;
+        if(!NextTwoPower<32, 128>(GemmNPerBlock))
+            break;
+        if(!NextTwoPower<4, 16>(GemmKPerBlock))
+            break;
+        if(!NextTwoPower<2, 4>(GemmMPerThread))
+            break;
+        if(!NextTwoPower<2, 4>(GemmNPerThread))
+            break;
+
+        return false;
+    } while(false);
+
+    return true;
+}
+
+std::string PerformanceImplicitGemmV4R4WrW::ToString() const
+{
+    std::ostringstream ss;
+    Serialize(ss);
+    return ss.str();
+}
+
+std::tuple<int, int, int>
+ConvHipImplicitGemmV4R4WrW::CalculateGemmSize(const ConvolutionContext& ctx)
+{
+    const auto n  = ConvolutionContextInterpreter::GetBatchN(ctx);
+    const auto c  = ConvolutionContextInterpreter::GetInputChannelC(ctx);
+    const auto k  = ConvolutionContextInterpreter::GetOutputChannelK(ctx);
+    const auto ho = ConvolutionContextInterpreter::GetOutputHeightHo(ctx);
+    const auto wo = ConvolutionContextInterpreter::GetOutputWidthWo(ctx);
+    const auto y  = ConvolutionContextInterpreter::GetFilterHeightY(ctx);
+    const auto x  = ConvolutionContextInterpreter::GetFilterWidthX(ctx);
+
+    const auto gemm_m = k;
+    const auto gemm_n =
+        c * y * x * (ctx.Is3d() ? ConvolutionContextInterpreter::GetFilterDepthZ(ctx) : 1);
+    const auto gemm_k =
+        n * ho * wo * (ctx.Is3d() ? ConvolutionContextInterpreter::GetOutputDepthDo(ctx) : 1);
+
+    return std::make_tuple(gemm_m, gemm_n, gemm_k);
+}
+
+bool ConvHipImplicitGemmV4R4WrW::IsApplicable(const ConvolutionContext& ctx) const
+{
+    if(ctx.direction.IsForward())
```

Comment:
Does it support Backward Data?

Reply:
No , I will create another PR to prevent it